ruby/lib/gherkin3/token_scanner.rb

```diff
@@ -11,15 +13,23 @@ module Gherkin3
   # to look for Gherkin keywords for the associated language. The keywords are defined 
   # in gherkin-languages.json.
   class TokenScanner
+    protected
+
+    attr_reader :io
```

Comment:
The scanner is able to scan a String or IO. Nothing else (especially not another scanner).

The parser is able to get tokens from a scanner, and for the sake of API friendliness, it can create a scanner out of a string.

Maybe the whole if statement should move to the parse method. I also don't like that a String can mean file path.